File: CODEFORCES/636 Div3/4.cpp
#include <bits/stdc++.h>
using namespace std;

#define PB push_back
#define F first
#define S second
#define MP make_pair

void solve()
{
    int n, k;
    cin >> n >> k;
    vector<int> arr(n);
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    unordered_map<int, int> mp; //cnt of pairs which need 0 changes to make x
    vector<pair<int, int>> vec;
    for (int i = 0; i < n / 2; i++)
    {
        mp[arr[i] + arr[n - i - 1]]++;
        vec.PB(MP(min(arr[i], arr[n - i - 1]), max(arr[i], arr[n - i - 1])));
    }

    vector<int> pre(2 * k + 10,0);
    for (auto it : vec)
    {
        int l = it.F + 1;
        int r = it.S + k;

        pre[l] += 1;
        pre[r+1] -= 1;
    }
    int sum = 0;
    for(int i = 0; i <= 2*k+10; i++){
        sum += pre[i];
        pre[i] = sum;
    }
    
    int ans = n;
    for(int x = 2; x <= 2*k; x++)
    {
        int cnt0 = mp[x];
        int cnt1 = pre[x]-cnt0;
        int cnt2 = n/2 - cnt0 - cnt1;

        ans = min(ans,cnt1+2*cnt2);
    }

    cout << ans << "\n";
}

int main()
{
    static const int _ = []() {
        ios::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
        return 0;
    }();

    int t;
    cin >> t;
    while (t--)
    {
        solve();
    }
    return 0;
}

File: CODEFORCES/636 Div3/4.py
def main():
    t = int(input())
    while t > 0:
        n, k = map(int, input().split())
        arr = [int(e) for e in input().split()]

        mp = {}
        vec = []
        for i in range(int(n/2)):
            if arr[i] + arr[n-i-1] not in mp:
                mp[arr[i]+arr[n-i-1]] = 0
            mp[arr[i]+arr[n-1-i]] += 1
            vec.append([min(arr[i], arr[n-i-1]), max(arr[i], arr[n-i-1])])

        pre = [0 for i in range(2*k+10)]

        for f, s in vec:
            l = f + 1
            r = s + k
            pre[l] += 1
            pre[r+1] -= 1

        sum = 0
        for i in range(0, 2*k+11):
            sum += pre[i]
            pre[i] = sum

        ans = n
        for x in range(2, 2*k+1):
            cnt0 = mp.get(x)
            if cnt0 == None:
                continue
            cnt1 = pre[x]-cnt0
            cnt2 = n/2-cnt0-cnt1
            ans = min(ans, cnt1+2*cnt2)

        print(ans)
        t -= 1

if __name__ == "__main__":
    main()
